cert/selfsigned_test.go

```diff
@@ -0,0 +1,65 @@
+package cert_test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/cert"
+)
+
+// TestIsOutdatedCert checks that we'll consider the TLS certificate outdated
+// if the ip addresses or dns names don't match.
+func TestIsOutdatedCert(t *testing.T) {
+	tempDir, err := ioutil.TempDir("", "certtest")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	certPath := tempDir + "/tls.cert"
+	keyPath := tempDir + "/tls.key"
+
+	extraIPs := []string{"1.1.1.1", "123.123.123.1", "199.189.12.12"}
+	extraDomains := []string{"home", "and", "away"}
+
+	// Generate TLS files with two extra IPs and domains.
+	err = cert.GenCertPair(
+		"lnd autogenerated cert", certPath, keyPath,
+		extraIPs[:2], extraDomains[:2],
+		cert.DefaultAutogenValidity,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// We'll attempt to check up-to-date status for all variants of 1-3
+	// number of IPs and domains.
+	for numIPs := 1; numIPs <= 3; numIPs++ {
```

Comment:
you could use `len(extraIPs)` and `len(extraDomains)` instead

Reply:
That is much better indeed!